server/database.go

```diff
@@ -765,18 +765,18 @@ func (s *Server) registerDatabaseRoutes(g *echo.Group) {
 			return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to update data source with ID %d", dataSourceID)).SetInternal(err)
 		}
 
-		if dataSourcePatch.SyncSchema {
-			// Refetch the instance to get the updated data source.
-			updatedInstance, err := s.store.GetInstanceByID(ctx, database.InstanceID)
-			if err != nil {
-				return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to fetch updated instance with ID %d", database.InstanceID)).SetInternal(err)
-			}
-			if err := s.syncEngineVersionAndSchema(ctx, updatedInstance); err != nil {
-				log.Warn("Failed to sync instance schema",
-					zap.Int("instance_id", updatedInstance.ID),
-					zap.Error(err))
-			}
+		// Refetch the instance to get the updated data source.
+		updatedInstance, err := s.store.GetInstanceByID(ctx, database.InstanceID)
+		if err != nil {
+			return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to fetch updated instance with ID %d", database.InstanceID)).SetInternal(err)
+		}
+		if _, err := s.syncInstance(ctx, updatedInstance); err != nil {
+			log.Warn("Failed to sync instance",
+				zap.Int("instance_id", updatedInstance.ID),
+				zap.Error(err))
 		}
+		// Sync all databases in the instance asynchronously.
+		instanceSyncChan <- updatedInstance
```

Comment:
move this inside syncInstance?

Reply:
The syncInstance() in sync schema runner doesn't need it. Let me have a follow-up to improve the sync schema runner.